astropy/coordinates/sky_coordinate.py

```diff
@@ -1163,6 +1163,46 @@ def spherical_offsets_to(self, tocoord):
         dlat = acoord.spherical.lat.view(Angle)
         return dlon, dlat
 
+    def spherical_offsets_by(self, d_lon, d_lat):
+        """
+        Computes the coordinates at the specified angular offsets from this
+        coordinate.
+
+        Parameters
+        ----------
+        d_lon : angle-like
+            The angular offset in the longitude direction (i.e. RA for
+            equatorial coordinates).
+        d_lat : angle-like
+            The angular offset in the latitude direction (i.e. Dec for
+            equatorial coordinates).
+
+        Returns
+        -------
+        newcoord : `~astropy.coordinates.SkyCoord`
+            The coordinates for the location that corresponds to offsetting by
+            ``d_lat`` in the latitude direction and ``d_lon`` in the longitude
+            direction.
+
+        Notes
+        -----
+        This internally uses `~astropy.coordinates.SkyOffsetFrame` to do the
+        transformation. For a more complete set of transform offsets, use
+        `~astropy.coordinates.SkyOffsetFrame` or `~astropy.wcs.WCS` manually.
+
+        See Also
+        --------
+        spherical_offsets_to : compute the angular offsets to another coordinate
+        directional_offset_by : offset a coordinate by an angle in a direction
+        """
+        if self.shape:
+            raise ValueError(
```

Comment:
I think in principle the code works for non-scalar coordinates, though my guess is that then the offsets have to dimensions equal to that of frame (given #8812).

Maybe add a comment that this can be removed if #8812 is implemented?

Reply:
Ah, I was thinking that it wouldn't work until #8812 is fixed, but you're right that it does work now if the shapes are identical. I think given that, fine to remove the error here!